src/router/response.rs

```diff
@@ -54,36 +61,69 @@ impl Response {
         self
     }
 
-    pub fn set_header(mut self, key: header::HeaderName, value: &'static str) -> Self {
+    pub fn set_header(mut self, key: header::HeaderName, value: &str) -> Self {
         match self.headers {
-            Some(ref mut x) => x.push((key, value)),
-            None => self.headers = Some(vec![(key, value)]),
+            Some(ref mut x) => x.push((key, value.to_string())),
+            None => self.headers = Some(vec![(key, value.to_string())]),
         };
         self
     }
 
     // Alias set_header method
-    pub fn with_header(self, key: header::HeaderName, value: &'static str) -> Self {
+    pub fn with_header(self, key: header::HeaderName, value: &str) -> Self {
         self.set_header(key, value)
     }
 
-    pub fn set_header_str(self, key: &'static str, value: &'static str) -> Self {
+    pub fn set_header_str(self, key: &'static str, value: &str) -> Self {
         self.set_header(
             header::HeaderName::from_bytes(key.as_bytes()).unwrap(),
             value,
         )
     }
 
     // Alias set_header_str method
-    pub fn with_header_str(self, key: &'static str, value: &'static str) -> Self {
+    pub fn with_header_str(self, key: &'static str, value: &str) -> Self {
         self.set_header_str(key, value)
     }
 
-    pub fn set_content_type(self, content_type: &'static str) -> Self {
+    pub fn set_content_type(self, content_type: &str) -> Self {
         self.set_header(header::CONTENT_TYPE, content_type)
     }
 
-    pub fn set_headers(mut self, headers: Vec<(header::HeaderName, &'static str)>) -> Self {
+    pub fn with_cookie(self, cookie: Cookie<'static>) -> Self {
+        self.set_cookie(cookie)
+    }
+
+    pub fn with_cookies(self, cookies: Vec<Cookie<'static>>) -> Self {
+        self.set_cookies(cookies)
+    }
+
+    pub fn with_cookie_raw(self, cookie: &str) -> Self {
+        self.set_header(header::SET_COOKIE, cookie)
+    }
+
+    pub fn set_cookie(mut self, cookie: Cookie<'static>) -> Self {
+        match self.cookies {
+            Some(ref mut x) => x.push(cookie),
+            None => self.cookies = Some(vec![cookie]),
+        }
+        self
+    }
+
+    pub fn set_cookies(mut self, mut cookies: Vec<Cookie<'static>>) -> Self {
+        match self.cookies {
+            Some(ref mut x) => x.append(&mut cookies),
+            None => self.cookies = Some(cookies),
+        }
+        self
+    }
+
+    pub fn set_headers(mut self, headers: Vec<(header::HeaderName, &str)>) -> Self {
```

Comment:
Shouldn't this take `&[Cookie<'static>]`?

Reply:
It set to `Vec` due to lifetime issue i guess. This is from @jk-gan #34.

Reply:
Oh